src/main.rs

```diff
@@ -19,7 +21,10 @@ extern crate log;
 /// Fast Port Scanner built in Rust.
 /// WARNING Do not use this program against sensitive infrastructure since the
 /// specified server may not be able to handle this many socket connections at once.
+/// - Discord https://discord.gg/rAnvBbg
+/// - GitHub https://github.com/RustScan/RustScan
 struct Opts {
+    // TODO default_value is a hack to make it an optional argument
```

Comment:
If we want `ip` as optional we need to do `Option<ip>`, but we have to adjust the code accordingly if we do that. 😄